tasks/download.sh

```diff
@@ -36,7 +36,16 @@ verify-file() {
 
 download() {
   printf '%s\n' "Downloading: ${1}"
-  curl -s -f -L -o "$2" "$1"
+  tmp_file=$(mktemp "peadm-download")
+  echo "Temporary file created at: ${tmp_file}"
+  curl -s -f -L -o ${tmp_file} "$1"
```

Comment:
We should make sure to check whether or not this curl command succeeds. If it doesn't succeed, we should `return 1`, even if somehow the tmpfile looks like a tarball.